tests/gui/modules/image-writer.spec.js

```diff
@@ -1,132 +1,99 @@
-      ImageWriterService = _ImageWriterService_
-    }))
+describe('Browser: imageWriter', function () {
+  describe('.flash()', function () {
+    describe('given a successful write', function () {
+      beforeEach(function () {
+        this.performWriteStub = m.sinon.stub(imageWriter, 'performWrite')
+        this.performWriteStub.returns(Bluebird.resolve({
+          cancelled: false,
+          sourceChecksum: '1234'
+        }))
+      })
 
-    describe('.flash()', function () {
-      describe('given a successful write', function () {
-        beforeEach(function () {
-          this.performWriteStub = m.sinon.stub(ImageWriterService, 'performWrite')
-          this.performWriteStub.returns($q.resolve({
-            cancelled: false,
-            sourceChecksum: '1234'
-          }))
-        })
+      afterEach(function () {
+        this.performWriteStub.restore()
+      })
 
-        afterEach(function () {
-          this.performWriteStub.restore()
+      it('should set flashing to false when done', function () {
+        flashState.unsetFlashingFlag({
+          cancelled: false,
+          sourceChecksum: '1234'
         })
 
-        it('should set flashing to false when done', function () {
-          flashState.unsetFlashingFlag({
-            cancelled: false,
-            sourceChecksum: '1234'
-          })
-
-          ImageWriterService.flash('foo.img', '/dev/disk2')
-          $rootScope.$apply()
+        imageWriter.flash('foo.img', '/dev/disk2').finally(() => {
           m.chai.expect(flashState.isFlashing()).to.be.false
         })
+      })
 
-        it('should prevent writing more than once', function () {
-          flashState.unsetFlashingFlag({
-            cancelled: false,
-            sourceChecksum: '1234'
-          })
+      it('should prevent writing more than once', function () {
+        flashState.unsetFlashingFlag({
+          cancelled: false,
+          sourceChecksum: '1234'
+        })
 
-          ImageWriterService.flash('foo.img', '/dev/disk2')
-          ImageWriterService.flash('foo.img', '/dev/disk2').catch(angular.noop)
-          $rootScope.$apply()
+        const writing = imageWriter.flash('foo.img', '/dev/disk2')
+        imageWriter.flash('foo.img', '/dev/disk2').catch(angular.noop)
+        writing.finally(() => {
           m.chai.expect(this.performWriteStub).to.have.been.calledOnce
         })
+      })
 
-        it('should reject the second flash attempt', function () {
-          ImageWriterService.flash('foo.img', '/dev/disk2')
-
-          let rejectError = null
-          ImageWriterService.flash('foo.img', '/dev/disk2').catch(function (error) {
-            rejectError = error
-          })
-
-          $rootScope.$apply()
+      it('should reject the second flash attempt', function () {
+        imageWriter.flash('foo.img', '/dev/disk2')
 
+        let rejectError = null
+        imageWriter.flash('foo.img', '/dev/disk2').catch(function (error) {
```

Comment:
I think we can now use fat arrows in most of these tests. Angular stuff heavily rely on `function`, and now that we're using `Bluebird` instead of `$q`, I think we can now use fat arrows.